File: ecbs-os/frontend/src/lib/clientManagementData.ts
export type ClientManagementData = {
  clients: ClientManagementClientRow[];
  clientKpis: ClientManagementKpi[];
  message: string;
  projects: ClientManagementProjectRow[];
  projectKpis: ClientManagementKpi[];
  selectedClient: ClientManagementSelectedClient;
  state: "data" | "no-data";
  updatedAt: string;
};

export type ClientManagementClientRow = {
  activeProjects: string;
  contractNumber: string;
  industry: string;
  joinedDate: string;
  name: string;
  sites: string;
  status: string;
  totalCapacity: string;
};

export type ClientManagementKpi = {
  detail: string;
  icon: string;
  label: string;
  tone: "blue" | "cyan" | "green" | "yellow";
  value: string;
};

export type ClientManagementProjectRow = {
  capacity: string;
  location: string;
  name: string;
  progress: string;
  siteType: string;
  startDate: string;
  status: string;
  targetCompletion: string;
};

export type ClientManagementSelectedClient = {
  accountManager: string;
  activeProjects: string;
  address: string;
  annualSavings: string;
  clientSince: string;
  completedProjects: string;
  contractNumber: string;
  currency: string;
  email: string;
  industry: string;
  legalName: string;
  mobile: string;
  name: string;
  phone: string;
  primaryContactName: string;
  primaryContactTitle: string;
  status: string;
  taxId: string;
  timeZone: string;
  totalCapacity: string;
  totalSites: string;
  website: string;
};

File: ecbs-os/backend/src/ECBS.Application/ScreenData/ClientManagementData.cs
namespace ECBS.Application.ScreenData;

public sealed record ClientManagementData(
    IReadOnlyList<ClientManagementClientRow> Clients,
    IReadOnlyList<ClientManagementKpi> ClientKpis,
    string Message,
    IReadOnlyList<ClientManagementProjectRow> Projects,
    IReadOnlyList<ClientManagementKpi> ProjectKpis,
    ClientManagementSelectedClient SelectedClient,
    string State,
    string UpdatedAt);

public sealed record ClientManagementClientRow(
    string ActiveProjects,
    string ContractNumber,
    string Industry,
    string JoinedDate,
    string Name,
    string Sites,
    string Status,
    string TotalCapacity);

public sealed record ClientManagementKpi(string Detail, string Icon, string Label, string Tone, string Value);

public sealed record ClientManagementProjectRow(
    string Capacity,
    string Location,
    string Name,
    string Progress,
    string SiteType,
    string StartDate,
    string Status,
    string TargetCompletion);

public sealed record ClientManagementSelectedClient(
    string AccountManager,
    string ActiveProjects,
    string Address,
    string AnnualSavings,
    string ClientSince,
    string CompletedProjects,
    string ContractNumber,
    string Currency,
    string Email,
    string Industry,
    string LegalName,
    string Mobile,
    string Name,
    string Phone,
    string PrimaryContactName,
    string PrimaryContactTitle,
    string Status,
    string TaxId,
    string TimeZone,
    string TotalCapacity,
    string TotalSites,
    string Website);
